### PR TITLE
[Merged by Bors] - Add cargo vendor test

### PR DESCRIPTION
**Pull request by bachp**
## Issue Addressed

This is related to #1926 and #1712.

## Proposed Changes

This PR adds a test that make sure that the used dependencies can be vendored.

Being able to vendor the dependencies is important for archival and repdroducibility purpose.
It's also required to package lighthouse for some Linux distributions. Specifically [NixOS](https://nixos.org/) and [Yocto](https://www.yoctoproject.org/).

## Additional Info

This PR only adds the test, it doesn't clean up the dependencies yet. That's why it is in draft.

### CODE REVIEW

**Comment by blacktemplar** [comment]
Thanks for the PR. As you already wrote, before we can merge this PR we need to fix the dependencies. Unfortunately we are blocked on some of those dependencies for the moment, see this issue: https://github.com/sigp/lighthouse/issues/1712. Hopefully we can resolve those as soon as possible and add in this test.

**Comment by bachp** [comment]
Looks like this is currently blocked by https://github.com/rust-lang/cargo/issues/8443. Which prevents cargo vendor from running in Github Actions.

**Comment by paulhauner** [comment]
> Looks like this is currently blocked by [rust-lang/cargo#8443](https://github.com/rust-lang/cargo/issues/8443). Which prevents cargo vendor from running in Github Actions.

Perhaps we can work around this by just re-defining `$HOME` to the target of the symlink? If there's no appetite for others to do this I might get around to it at somepoint :)

**Comment by bachp** [comment]
With the workaround of https://github.com/rust-lang/cargo/issues/8443#issuecomment-951316266 the job is finally working, but there is an issue again that `multistream-select v0.10.4` is vendored from multiple sources.

**Comment by bachp** [comment]
In unstable the vendor issues are fixed. @paulhauner tests are now passing, so this is finally ready to be merged.

**Review by paulhauner — APPROVED** [review]
Ooh, how exciting! :tada: 

Thanks for sticking with this one for the long run @bachp :pray: 

bors r+

**Pull request closed without merging**
